egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,299 @@
+#!/usr/bin/env python
+# Copyright 2016  Tom Ko
+# Apache 2.0
+# script to generate reverberated data
+
+# we're using python 3.x style print but want it to work in python 2.x,
+from __future__ import print_function
+import argparse, glob, math, os, random, sys, warnings, copy, imp, ast
+
+train_lib = imp.load_source('ntl', 'steps/nnet3/nnet3_train_lib.py')
+
+class list_cyclic_iterator:
+  def __init__(self, list, random_seed = 0):
+    self.list_index = 0
+    self.list = list
+    random.seed(random_seed)
+    random.shuffle(self.list)
+
+  def next(self):
+    item = self.list[self.list_index]
+    self.list_index = (self.list_index + 1) % len(self.list)
+    return item
+
+
+def GetArgs():
+    # we add compulsary arguments as named arguments for readability
+    parser = argparse.ArgumentParser(description="Generate corrupted data"
```

Comment:
"Reverberate the data directory with an option to add isotropic and point source noises"